### ISSUE TITLE
download updates from local network

### ISSUE THREAD

**Issue by noah1510** [linuxmint/mintupdate#646]
tldr: The idea is basically [this](https://www.digitalcitizen.life/how-set-windows-10-get-updates-local-network-internet) function implemented using warpinators protocol with the settings options in the update manager

The goal is to speed up the download speed of updates by downloading the newer package versions from other mint computers on the (local) network.
There should be two options in the update manager 'share downloaded updates on the local network' and 'download updates from the local network'.
This way only one computer on the local network has to download the packages and the other computers can use that one as a source (if they do the signature should always be checked), speeding up the download especially on networks with slower internet connection and reduce the total internet usage (useful if there are data limits by your internet provider).

As mentioned this should be part of the update manager but ideally the backend should be written in a way that is not dependent on any specific package manager, in theory also allowing the same system to be used for flatpaks, snaps, arch packages ,etc..
For discovery and file transfer warpinator can be utilized (or at least the protocol).
To make sure only valid packages are sent, the signature should always be checked and the version should be downloaded from the official repositories.

This should also be as easy as possible so that any user of mint can easily benefit from it while having roughly the same amount of security.

**Comment by xenopeek** [comment]
Thanks for making a suggestion for Update Manager. It's logged as a feature request for future consideration. See the [workflow](https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html#why-are-ideas-and-feature-requests-closed) for why it is closed here.

If you have a need for this functionality now you can do so by setting up a local apt cache or local apt mirror on one of your computers (using apt-cacher or apt-mirror programs) and configuring your other computers to use the apt repository served from that computer instead of any of the public mirrors. There are tutorials out there for both. If you would want to use either and need help with that you can ask on the Linux Mint forums.
